### DISCUSSION TITLE
Persistent DB Data & Container Rebuilds

### DISCUSSION THREAD

**Discussion by metaskills** [community/community#7374] [Codespaces] [answered]
When using `dockerComposeFile` it would be common to have additional services like like a MySQL database. Most services have a way to use Docker volumes to put data in a place that would be persisted. I've been curious how to do this with Codespaces.

**Comment by metaskills** [comment, marked as answer]
I have found the following works because the top level host that runs your `service` keeps the contents of this directory during rebuilds. I did try other folders like `/var/lib/docker/codespacemount/workspace/${service}` or even `/var/lib/docker/codespacemount/.persistedshare` but found those did not work for one reason on another. So the question is, how does the Codespaces team (or others) feel about this solution? 


```dockerfile
mysql57:
  image: mysql:5.7
  restart: unless-stopped
  environment:
    MYSQL_ROOT_PASSWORD: root
  volumes: ['/var/lib/docker/codespacemount/data-mysql:/var/lib/mysql']
```

**Comment by misner** [comment]
@metaskills how did you manage to persists local/dev database indeed as when you rebuild a codespace, any Docker containers, images, volumes, and caches are cleaned (=wiped out of any data ([Source](https://docs.github.com/en/codespaces/developing-in-codespaces/codespaces-lifecycle#rebuilding-a-codespace)) ?
